src/tools.ts

```diff
@@ -28,6 +17,123 @@ function getRangesFromDecorations (
     .map((decoration) => decoration.range)
 }
 
+function minusRanges (uniqueRange: monaco.Range, ranges: monaco.Range[]): monaco.Range[] {
+  const newRanges: monaco.Range[] = []
+  let lastEndLineNumber = uniqueRange.startLineNumber
+  let lastEndColumn = uniqueRange.startColumn
+
+  for (const range of ranges) {
+    const newRange = new monaco.Range(lastEndLineNumber, lastEndColumn, range.startLineNumber, range.startColumn)
```

Comment:
nit: you can use a monaco.Position instead of the lineNumber and column

Reply:
7ee4792